### DISCUSSION TITLE
How I can use the converted xaml  in WPF with vector

### DISCUSSION THREAD

**Discussion by yugiking** [ElinamLLC/SharpVectors#179] [Q&A]
hi, 
Thanks for this great tool!
And I have a problem as below:
I use the FileSvgConverter to convert my svg file to xaml, and insert this xaml content to my WPF window(default file name is MainWindow.xaml),  code as below:
 
```
<Image>    
    <Image.Source>
        <DrawingImage>
            <DrawingImage.Drawing>
                <DrawingGroup .../>      //this is the content of the converted xaml
            </DrawingImage.Drawing>
        </DrawingImage>
    </Image.Source>
</Image>
```
The image can show out, but it's not vector, it means that I can't select the content within the image, so my question is how I can use the converted xaml so it can show in WPF with vector? by using other control instead of  "Image" ?
I have this requirement because I need to export my WPF window to a XPS file and the XPS file's content should be accessible

**Comment by paulushub** [comment]
@yugiking Since you are using `Image`, which is a static image, you cannot get it to be interactive. If need to convert to a file and then later display that, use the `SvgDrawingCanvas`, you can implement hit-testing features and make it interactive.

**Reply by yugiking** [reply]
Thanks your reply @paulushub, is the "SvgDrawingCanvas" only defined in this project? or a common control? It would be much helpful if you can provide me an example on how to use this control

**Reply by paulushub** [reply]
@yugiking Many samples use this control. See
https://github.com/ElinamLLC/SharpVectors/blob/master/Samples/WpfSvgTestBox/SvgPage.xaml

`SvgDrawingCanvas` is contained in the `SharpVectors.Runtime` project, it is a standalone project (independent of other SharpVectors project) so you can simply distribute this DLL with your application if you are only shipping the converted XAML files, or take the `SvgDrawingCanvas.cs` file and customize it to your needs.

**Reply by yugiking** [reply]
Thanks your reply again but I am sorry that still not sure how to use the SvgDrawingCanvas ref to the sample you mentioned,  I append the xaml content (the root tag of DrawingGroup) to the SvgDrawingCanvas but not work:

```
    <svg:SvgDrawingCanvas x:Name="svgDrawing" Background="Transparent">
        <DrawingGroup>
             //this is the content of the converted xaml
        </DrawingGroup>
    </svg:SvgDrawingCanvas>
```

what other tag I should add before I can append the DrawingGroup content to the SvgDrawingCanvas ?

**Reply by paulushub** [reply]
@yugiking Look at the code behind, `SvgPage.cs`. This generic control does not support binding, that is implemented by controls that extend it like `SvgCanvas` or controls that wrap it like `SvgViewbox`.
So remove the `DrawingGroup` tag in your code and use the `LoadDiagrams/UnloadDiagrams` methods to load the files or `RenderDiagrams` to render the `DrawingGroup` if you are loading it yourself.

```xml
<svg:SvgDrawingCanvas x:Name="svgDrawing" Background="Transparent"/>
```
In your code behind...
```csharp
svgDrawing.LoadDiagrams("file path")
```
